### PR TITLE
ST_DumpSegments

### PR DESCRIPTION
**Pull request by kalenikaliaksandr**
[https://lists.osgeo.org/pipermail/postgis-users/2013-October/037934.html](https://lists.osgeo.org/pipermail/postgis-users/2013-October/037934.html) implemented in C

### CODE REVIEW

File: postgis/postgis.sql.in [hunk 1]
```diff
@@ -1960,6 +1960,13 @@ CREATE OR REPLACE FUNCTION ST_DumpPoints(geometry)
 	LANGUAGE 'c' IMMUTABLE STRICT PARALLEL SAFE
 	_COST_HIGH;
 
+-- Availability: 3.2.0
```
Comment (kalenikaliaksandr) [inline postgis/postgis.sql.in:1963]: too optimistic?

Comment (Komzpa) [reply postgis/postgis.sql.in:1963]: I believe it's ok

File: postgis/lwgeom_dumpsegments.c [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+		POINT4D pt_start, pt_end;
+		POINTARRAY *segment_pa;
+		LWLINE *segment;
+
+		node = &state->stack[state->stacklen - 1];
+		lwgeom = node->geom;
+
+		if (lwgeom->type == LINETYPE || lwgeom->type == POLYGONTYPE)
+		{
+			if (lwgeom->type == LINETYPE)
```
Comment (Komzpa) [inline postgis/lwgeom_dumpsegments.c:159]: Original had triangle too. DumpSegments is also used around constrained Delaunay triangulations, and that means that triangles/TINs can be the thing you try to dump as segments too.

**Review by Komzpa — CHANGES_REQUESTED** [review]

File: postgis/lwgeom_dumpsegments.c [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **********************************************************************
+ *
+ * ^copyright^
```
[the remark this answers is not in the document]
Comment (kalenikaliaksandr) [reply postgis/lwgeom_dumpsegments.c]: Kontur? :)

File: postgis/lwgeom_dumpsegments.c [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+#include "funcapi.h"
+
+#include "../postgis_config.h"
+#include "lwgeom_pg.h"
+
+#include "access/htup_details.h"
+
+#include "liblwgeom.h"
+
+Datum LWGEOM_dumpsegments(PG_FUNCTION_ARGS);
```
[the remark this answers is not in the document]
Comment (Komzpa) [reply postgis/lwgeom_dumpsegments.c:37]: It's a smaller issue than having the code duplicated IMHO. We can rename the file.

**Pull request merged**